modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -1346,6 +1392,183 @@ void WindowContextTop::notify_window_move() {
     }
 }
 
+void WindowContextTop::show_system_menu(int x, int y) {
+    GdkDisplay* display = gdk_display_get_default();
+    if (!display) {
+        return;
+    }
+
+    GdkSeat* seat = gdk_display_get_default_seat(display);
+    GdkDevice* device = gdk_seat_get_pointer(seat);
+    if (!device) {
+        return;
+    }
+
+    gint rx = 0, ry = 0;
+    gdk_window_get_root_coords(gdk_window, x, y, &rx, &ry);
+
+    GdkEvent* event = (GdkEvent*)gdk_event_new(GDK_BUTTON_PRESS);
+    GdkEventButton* buttonEvent = (GdkEventButton*)event;
+    buttonEvent->x_root = rx;
+    buttonEvent->y_root = ry;
+    buttonEvent->window = (GdkWindow*)g_object_ref(gdk_window);
+    buttonEvent->device = (GdkDevice*)g_object_ref(device);
+
+    gdk_window_show_window_menu(gdk_window, event);
+    gdk_event_free(event);
```

Comment:
The `seat` concept was introduced in gtk 3.20  and `gdk_window_show_window_menu` was in 3.14, and we require 3.8.  
I would agree that it's time to bump it up.

Reply:
Is there anything we need to change here?

Reply:
I'm okay with bumping the required version. It seems like 3.8 was released 12 years ago, and we would be requiring a version that was released 9 years ago. Sounds fair.

Reply:
Our production build system is Oracle Linux 8, which has GTK 3.22.

The oldest system I now have access to is Ubuntu 16.04 (yeah, I know...it's probably time to retire it), which has GTK 3.18. I build JavaFX using GCC 14.2 (via a devkit) and compile against a GTK 3.22 header files and library. I'm able to run this PR and use the EXTENDED stage style with a HeaderBar, but when I try to use the context menu it crashes:

```
java: symbol lookup error: build/sdk/lib/libglassgtk3.so: undefined symbol: gdk_display_get_default_seat
```

We should make a deliberate decision as to whether or not to bump the minimum. I think bumping the minimum to 3.20 might be reasonable thing to, although I'd want to do it separately from this PR and provide a release note for that.

Reply:
I mis-remembered the GTK version in OL7. I just tried it on an OL 7 VM I still had lying around, and it also has GTK 3.22 and using this PR, the MonkeyTester with EXTENDED stage style and a HeaderBar runs just fine, including the context menu.

> I think providing a release note that the new extended stage style requires GTK 3.20 sounds reasonable.

I was more thinking that we would file a new issue to bump the minimum to 3.20 and provide a release note using that bug that JavaFX requires 3.20 as a minimum. If we end up not doing that, then a release note that the extended stage style needs 3.20 would be good.

Reply:
I can work on the Gtk version bump.